src/main/java/tdl/datapoint/video/security/TokenSigningService.java

```diff
@@ -0,0 +1,43 @@
+package tdl.datapoint.video.security;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.crypto.Mac;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+
+public class TokenSigningService {
+    private static final Logger log = LoggerFactory.getLogger(TokenSigningService.class);
+
+    private Mac mac;
+
+    public TokenSigningService(String encryptionKey) throws NoSuchAlgorithmException, InvalidKeyException {
+        mac = Mac.getInstance("HmacMD5");
+        mac.init(asSecretKey(encryptionKey));
+    }
+
+    private SecretKey asSecretKey(String encryptionKey) {
+        byte[] encryptionKeyAsBytes = encryptionKey.getBytes();
+        return new SecretKeySpec(encryptionKeyAsBytes, 0, encryptionKeyAsBytes.length, "AES");
+    }
+
+    private String sign(String token) {
```

Comment:
This is not signing as the message cannot be recovered.

Reply:
Will change that

Reply:
Done, we can discuss further if changes are needed